sdk/core/Azure.Core/src/ProcessingContext.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Contains information related to the processing of the <see cref="HttpMessage"/>.
+    /// </summary>
+    public readonly struct ProcessingContext
```

Comment:
Josh and I measured this just to validate it does perform better from an allocation perspective over modeling this as a class with local properties.

```
|     Method |     Mean |   Error |  StdDev | Ratio | RatioSD |  Gen 0 | Allocated |
|----------- |---------:|--------:|--------:|------:|--------:|-------:|----------:|
| StructType | 177.1 ns | 3.40 ns | 3.78 ns |  0.97 |    0.04 | 0.0057 |      48 B |
|  ClaseType | 182.4 ns | 3.67 ns | 3.93 ns |  1.00 |    0.00 | 0.0086 |      72 B |
```

Reply:
24 bytes savings! :-)